core/server/utils/image-size-from-url.js

```diff
@@ -51,71 +48,56 @@ module.exports.getImageSizeFromUrl = function getImageSizeFromUrl(imagePath) {
+        ).then(function (response) {
+            try {
+                dimensions = sizeOf(response.body);
 
-        request = requestHandler.get(options, function (res) {
-            var chunks = [];
+                imageObject.width = dimensions.width;
+                imageObject.height = dimensions.height;
 
-            res.on('data', function (chunk) {
-                chunks.push(chunk);
-            });
-
-            res.on('end', function () {
-                if (res.statusCode === 200) {
-                    try {
-                        dimensions = sizeOf(Buffer.concat(chunks));
-
-                        imageObject.width = dimensions.width;
-                        imageObject.height = dimensions.height;
-
-                        return resolve(imageObject);
-                    } catch (err) {
-                        return reject(new errors.InternalServerError({
-                            code: 'IMAGE_SIZE',
-                            err: err,
-                            context: imagePath
-                        }));
-                    }
-                } else if (res.statusCode === 404) {
-                    return reject(new errors.NotFoundError({
-                        message: 'Image not found.',
-                        code: 'IMAGE_SIZE',
-                        statusCode: res.statusCode,
-                        context: imagePath
-                    }));
-                } else {
-                    return reject(new errors.InternalServerError({
-                        message: 'Unknown Request error.',
-                        code: 'IMAGE_SIZE',
-                        statusCode: res.statusCode,
-                        context: imagePath
-                    }));
-                }
-            });
-        }).on('socket', function (socket) {
-            if (timeout) {
-                socket.setTimeout(timeout);
-
-                /**
-                 * https://nodejs.org/api/http.html
-                 * "...if a callback is assigned to the Server's 'timeout' event, timeouts must be handled explicitly"
-                 *
-                 * socket.destroy will jump to the error listener
-                 */
-                socket.on('timeout', function () {
-                    request.abort();
-                    socket.destroy(new Error('Request timed out.'));
-                });
+                return resolve(imageObject);
+            } catch (err) {
+                return reject(new errors.InternalServerError({
+                    code: 'IMAGE_SIZE',
+                    err: err,
+                    context: imagePath.href
+                }));
+            }
+        }).catch(function (err) {
+            if (err.statusCode === 404) {
+                return reject(new errors.NotFoundError({
+                    message: 'Image not found.',
+                    code: 'IMAGE_SIZE',
+                    statusCode: err.statusCode,
+                    context: err.url
```

Comment:
Hmm... in my tests, `err.url` has always been populated by `got`, but seeing [here](https://github.com/sindresorhus/got#errors) that is says "if available", I probably should return a conditional just to make sure.